src/librustc_data_structures/slice_ext.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+pub trait SliceExt<T> {
+    type Item;
+
+    fn pick2(&mut self, a: usize, b: usize) -> (&mut T, &mut T);
+}
+
+impl<T> SliceExt<T> for [T] {
+    type Item = T;
+
+    fn pick2(&mut self, a: usize, b: usize) -> (&mut T, &mut T) {
```

Comment:
Let's add a comment here?

Something like:

---

Return mutable references to two distinct elements, `a` and `b`. Panics if `a == b`.

---

Also, maybe we could call it `get2_mut`? (similar to `get_mut`?)

Reply:
At least I think we should add a `_mut` prefix

Reply:
Let's not call it `get2` since it doesn't return an `Option` (as `get` does).